website/api/controllers/entrance/signup.js

```diff
@@ -100,46 +104,46 @@ the account verification message.)`,
     let fleetSandboxDemoKey = await sails.helpers.strings.uuid();
 
     // Send a POST request to the cloud provisioner API
-    let cloudProvisionerResponse = await sails.helpers.http.post(
+    let cloudProvisionerResponseData = await sails.helpers.http.post(
       'https://sandbox.fleetdm.com/new',
       { // Request body
         'name': firstName + ' ' + lastName,
         'email': newEmailAddress,
-        'password': fleetSandboxDemoKey,
+        'password': fleetSandboxDemoKey, //« this provisioner API was originally designed to accept passwords, but rather than specifying the real plaintext password, since users always access Fleet Sandbox from their fleetdm.com account anyway, this generated demo key is used instead to avoid any confusion
         'sandbox_expiration': new Date(fleetSandboxExpiresAt).toISOString(), // sending expiration_timestamp as an ISO string.
       },
       { // Request headers
         'Authorization':sails.config.custom.cloudProvisionerSecret
       }
     )
     .timeout(5000)
-    .intercept('non200Response', ()=>{
-      // If we recieved a non-200 response from the Cloud Provisioner API, we'll throw a 500 error.
-      throw new Error('When attempting to provision a new user\'s Fleet Sandbox instance, the Cloud provisioner gave a non 200 response. The incomplete user record has not been saved in the database, and the user will be asked to try signing up again.');
+    .intercept(['requestFailed', 'non200Response'], (err)=>{
+      // If we recieved a non-200 response from the cloud provisioner API, we'll throw a 500 error.
+      return new Error('When attempting to provision a new user\'s Fleet Sandbox instance, the cloud provisioner gave a non 200 response. The incomplete user record has not been saved in the database, and the user will be asked to try signing up again. Raw response received from provisioner: '+err.stack);
```

Comment:
I overlooked this before, but similar to the other spot in this file:

> When attempting to provision a new user\'s Fleet Sandbox instance

Which user?

It would be helpful to include the email address of the user here too (similar to https://github.com/fleetdm/fleet/pull/6380#discussion_r929567471)

Reply:
In case it saves you time:  (I'll leave it to you to edit so I don't mess you up)

```js
return new Error('When attempting to provision a new Fleet Sandbox instance for a new user who just signed up ('+emailAddress+'), the cloud provisioner gave a non 200 response. The incomplete user record has not been saved in the database, and the user will be asked to try signing up again. Raw response received from provisioner: '+err.stack);
```